core/client-entry.ts

```diff
@@ -99,12 +103,9 @@ const invokeClientEntry = async () => {
         }
       }))
     })
-    // Mounting app
-    if (!RouterManager.isRouteDispatched()) {
-      RouterManager.addDispatchCallback(() => {
-        app.$mount('#app')
-      })
-    } else {
+    // if route is dispatched and there is no pending in router history
+    // if there is stil pending then 'beforeResolve' will handle app mount, otherwise 'beforeResolve' will not trigger
+    if (RouterManager.isRouteDispatched() && !(router as any).history.pending) {
```

Comment:
I guess this should be combined with the condition from the line 68 into single one for devs to make it easier to get what these conditions are doing :)

Reply:
I made update :) Also moved if in onReady callback on top of it.